### PR TITLE
Bump Jest to v27

### PR DESCRIPTION
**Pull request by mcmire**
## Description

The `mocked` utility function from `ts-jest`, which we commonly use, was moved into Jest in v27.

## Changes

N/A

## References

This is a blocker for https://github.com/MetaMask/core/issues/1197, as we want to update the NetworkController tests to use `jest.mocked` instead of `mocked` from `ts-jest/utils`.

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Comment by mcmire** [comment]
@legobeat Ah, thanks! I've updated this PR to bump Jest to only v27.

File: packages/assets-controllers/src/CurrencyRateController.test.ts
Comment (mcmire) [inline packages/assets-controllers/src/CurrencyRateController.test.ts:47]: Changed because Jest v27 switches to the "modern" fake timers by default and these must mock more things related to timers than the legacy fake timers, because now any of the tests that involve HTTP requests are timing out. I suspect because this is because [`node-fetch` calls `setTimeout` on the request object during the request-response flow](https://github.com/node-fetch/node-fetch/blob/7b86e946b02dfdd28f4f8fca3d73a022cbb5ca1e/src/index.js#L139), but I didn't look beyond that.

File: packages/transaction-controller/jest.config.js
Comment (mcmire) [reply packages/transaction-controller/jest.config.js:19]: Hmm, that's fair. Let me dive deeper into the coverage reports.

Comment (Gudahtt) [reply packages/transaction-controller/jest.config.js:19]: For the preferences controller tests, the branch coverage went from 22/24 to 13/15. Here in the transactions controller, branch coverage went from 217/264 to 185/229

The reports seem to show the same gaps, roughly. I did a visual comparison and they were nearly identical. The one difference is that [the else case for this condition](https://github.com/MetaMask/core/blob/main/packages/transaction-controller/src/TransactionController.ts#L680) used to be flagged as uncovered, but now it isn't.

I did a quick search in the Jest issue tracker but couldn't find any reference to this.

This is curious, but I don't think it needs to block this PR.

File: packages/assets-controllers/src/CurrencyRateController.test.ts
Comment (Gudahtt) [reply packages/assets-controllers/src/CurrencyRateController.test.ts:47]: More information on this: https://github.com/nock/nock/issues/2200

It seems that using the legacy implementation is the best solution for now. Jest v28 offers more options though. When we update to v28 we should consider using the modern timers but configuring it to leave whatever we need here unmocked.

**Review by Gudahtt — COMMENTED** [review]
Looks great! I can approve once the conflicts are resolved

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**
